### PR TITLE
feat: menu handler

### LINKED ISSUE #2: Use ArangoDb for peristence
**Is your feature request related to a problem? Please describe.**
The access to the ArangoDB should be provided for persistence.

**Describe the solution you'd like**
An arango client is used for convenience.

**Describe alternatives you've considered**
The following client should be used: https://docs.rs/arangors/0.4.5/arangors/

**Additional context**
None

### LINKED ISSUE #3: Use BDD for testing
**Is your feature request related to a problem? Please describe.**
All features provided by the server should be reflected in feature scenarios.

**Describe the solution you'd like**
For this purpose, a BDD tool like cucumber is used in order to create behavior driven tests.

**Describe alternatives you've considered**
The following tool might be useful: https://github.com/bbqsrc/cucumber-rust

**Additional context**
None

### LINKED ISSUE #4: Research on a matching architecture
**Is your feature request related to a problem? Please describe.**
This server, even though written in an unmature framework, should follow a set of design principles.

**Describe the solution you'd like**
The goal of this research should provide insights into specific design decisions (Module split, responsibility, monitoring, logging etc.)

**Describe alternatives you've considered**
The following articles were already found:
1. https://medium.com/@ilegra/building-a-microservice-with-rust-ef9641cf2331
2. https://actix.rs/
3. https://medium.com/digitalfrontiers/microservices-in-rust-with-actix-a61bb62fee80

**Additional context**
None

### LINKED ISSUE #6: Add menu handler
**Is your feature request related to a problem? Please describe.**
The bistro server should provide a possibility to manage menus.

**Describe the solution you'd like**
Serve the following routes
```
GET /menu/<timeRange>
RETURNS List<MenuId>

GET /menu/<MenuId>
RETURNS Menu
```

**Describe alternatives you've considered**
Not yet

**Additional context**
None

### PR DESCRIPTION
**Pull request by ansgarS**
## Rules for creators (remove these after creation)
- What's the story behind this PR?
- Every PR must have description of Impact / Risks / Unknowns

## Rules for reviewers
- Reject large PRs if it lacks detailed description / story behind it
- Large PRs may be done pairing with author
- "Approve" only if you approve of functional impact
- "Comment" only if you just did a code-style review
- Reject PRs with insufficient test cases
- **Meaningful rejections make us better**
- Optional: Check for cyclic maven dependencies

## Description
This PR:
- serves all menu routes with rocket rs

## Review
- [ ] functional review
- [ ] codestyle review

## Risks 
*What could potentially happen due to these changes*

## Impact 
Improved navigation UX for mobile user
Less time needed to find related categories
Lower exit rate
More returning user

## Unknowns
- Not sure if ... is the correct solution. If you can come up with a better approach…

## Notes
Follow-Up ticket … created to clean … up.

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+    }
+}
+
+/// Index Route handler that provides information about
+/// the API server.
+///
+/// Each handler will respond with the ApiResponse, that
+/// is building a Rest response in the background. The
+/// status code of the API response is mapped into an
+/// http status code und the JSON is used as the
```
Comment (AliciaOh) [inline src/main.rs]:
```suggestion
/// http status code and the JSON is used as the
```

File: src/main.rs [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+        }),
+        status: Status::NotFound
+    }
+}
+
+/// Index Route handler that provides information about
+/// the API server.
+///
+/// Each handler will respond with the ApiResponse, that
+/// is building a Rest response in the background. The
```
Comment (AliciaOh) [inline src/main.rs]:
```suggestion
/// is building a REST response in the background. The
```

File: src/menu/repository.rs [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+    let aql = AqlQuery::builder()
+        .query("FOR menu IN @@collection RETURN menu")
+        .bind_var("@collection", config.menu_collection)
+        .build();
+
+    db.aql_query(aql).await
+}
+
+/// Queries menus that are served within the given time range
+/// Returns an empty list of no menu is found
```
Comment (AliciaOh) [inline src/menu/repository.rs]:
```suggestion
/// Returns an empty list if no menu is found
```

File: src/middleware/api_response.rs [hunk 4]
```diff
@@ -0,0 +1,27 @@
+use rocket::http::{ContentType, Status};
+use rocket::response::Responder;
+use rocket::{response, Request, Response};
+use rocket_contrib::json::JsonValue;
+
+/// Generic data model for all api responses
+/// It is split into the json body and a http
```
Comment (AliciaOh) [inline src/middleware/api_response.rs]:
```suggestion
/// It is split into the JSON body and a HTTP
```

File: src/middleware/api_response.rs [hunk 5]
```diff
@@ -0,0 +1,27 @@
+use rocket::http::{ContentType, Status};
+use rocket::response::Responder;
+use rocket::{response, Request, Response};
+use rocket_contrib::json::JsonValue;
+
+/// Generic data model for all api responses
```
Comment (AliciaOh) [inline src/middleware/api_response.rs]:
```suggestion
/// Generic data model for all API responses
```

File: src/middleware/api_response.rs [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+///
+#[derive(Debug)]
+pub struct ApiResponse {
+    pub json: JsonValue,
+    pub status: Status,
+}
+
+/// Implementation of the responder trait for
+/// bistro API responses. It always expects
+/// an http status code and a json body.
```
Comment (AliciaOh) [inline src/middleware/api_response.rs]:
```suggestion
/// an HTTP status code and a JSON body.
```

File: src/middleware/date_query.rs [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+                    _ => (from, to)
+                }
+            });
+
+        let from = from_param.ok_or("Query param 'from' is missing")?.value.url_decode().map_err(|_| "Query param 'from' is malformed")?;
+        let to = to_param.ok_or("Query param 'to' is missing")?.value.url_decode().map_err(|_| "Query param 'to' is malformed")?;
+
+        Ok(DateRangeQueryParam {
+            from: NaiveDate::parse_from_str(&from, "%Y-%m-%d").map_err(|_| "Query param 'from' has to be in format YYYY-MM-DD")?,
+            to: NaiveDate::parse_from_str(&to, "%Y-%m-%d").map_err(|_| "Query param 'from' has to be in format YYYY-MM-DD")?,
```
Comment (AliciaOh) [inline src/middleware/date_query.rs]:
```suggestion
            to: NaiveDate::parse_from_str(&to, "%Y-%m-%d").map_err(|_| "Query param 'to' has to be in format YYYY-MM-DD")?,
```

**Review by AliciaOh — COMMENTED** [review]
I still need to go through your changes in more detail, but it already looks pretty good to me! Good job!

File: src/middleware/date_query.rs [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+#[derive(Copy, Clone, Debug)]
+pub struct DateRangeQueryParam {
+    pub from: NaiveDate,
+    pub to: NaiveDate
+}
+
+impl<'v> FromQuery<'v> for DateRangeQueryParam {
+    type Error = &'static str;
+
+    fn from_query(query: Query<'v>) -> Result<Self, Self::Error> {
```
Comment (RouHim) [inline src/middleware/date_query.rs]: Query is no longer available on master branch.

Comment (ansgarS) [reply src/middleware/date_query.rs]: Resolved with FromForm

**Comment by ansgarS** [comment]
@RouHim Please retry locally after running `cargo update -p rocket`

File: src/middleware/date_query.rs (hunk 7 above)
Comment (ansgarS) [reply src/middleware/date_query.rs]: Parsing of date query strings was rewritten

**Comment by ansgarS** [comment]
I would merge this PR if no further discussions arise this weekend

**Pull request merged**